intellij/src/saros/intellij/runtime/IntellijUISynchronizer.java

```diff
@@ -1,46 +1,42 @@
 package saros.intellij.runtime;
 
-import com.intellij.openapi.application.Application;
 import com.intellij.openapi.application.ApplicationManager;
-import com.intellij.openapi.application.ModalityState;
 import com.intellij.openapi.progress.ProcessCanceledException;
 import org.apache.log4j.Logger;
 import saros.synchronize.UISynchronizer;
 
 /**
- * Class implements the {@link UISynchronizer} with {@link Application#invokeLater(Runnable)} and
- * {@link Application#invokeAndWait(Runnable, ModalityState)}.
+ * Class implements the {@link UISynchronizer} by executing the given runnable on the event
+ * dispatcher thread (EDT).
+ *
+ * @see EDTExecutor
  */
 public class IntellijUISynchronizer implements UISynchronizer {
 
   private static final Logger log = Logger.getLogger(IntellijUISynchronizer.class);
 
   @Override
   public void asyncExec(Runnable runnable) {
-    exec(runnable, true);
+    try {
+      EDTExecutor.invokeLater(runnable);
+
+    } catch (ProcessCanceledException e) {
+      log.error("Synchronous execution on EDT interrupted - " + runnable, e);
```

Comment:
```suggestion
      log.error("Asynchronous execution on EDT interrupted - " + runnable, e);
```

Reply:
Is this a possible scenario? I would expect that invokeLater directly returns and does not throw a ProcessCanceledException.

Reply:
This was just copy-pasted (badly) from the old logic. And no, it does not seem like the async execution can throw such an exception. Will remove the check.